hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/TrashPolicyDefault.java

```diff
@@ -157,14 +163,37 @@ public boolean moveToTrash(Path path) throws IOException {
       } catch (FileAlreadyExistsException e) {
         // find the path which is not a directory, and modify baseTrashPath
         // & trashPath, then mkdirs
+        if (moveToTrashForTests) {
+          try {
+            Thread.sleep(5000);
```

Comment:
Can we use a more deterministic approach to test the code, instead of a time based one?
Like CountDownLatch?